af/fastpath/fastpath/core.py

```diff
@@ -350,52 +351,599 @@ def match_fingerprints(measurement):
     return matches
 
 
-@metrics.timer("score_measurement")
-def score_measurement(msm, matches):
-    """Calculate measurement scoring. Returns a summary dict
+def truekeys(d, keys):
+    """Check for values set to True in a dict"""
+    if isinstance(keys, str):
+        keys = (keys,)
+    for k in keys:
+        if d.get(k, None) != True:
+            return False
+
+    return True
+
+
+def falsekeys(d, keys):
+    """Check for values set to True in a dict"""
+    if isinstance(keys, str):
+        keys = (keys,)
+    for k in keys:
+        if d.get(k, None) != False:
+            return False
+
+    return True
+
+
+def nonekeys(d, keys):
+    """Check for values set to None in a dict"""
+    if isinstance(keys, str):
+        keys = (keys,)
+    for k in keys:
+        if d.get(k, True) != None:
+            return False
+
+    return True
+
+
+def logbug(id: int, desc: str, msm: dict):
+    """Log unexpected measurement contents, possibly due to a bug in the probe
+    The id helps locating the call to logbug()
     """
-    # Extract interesting fields
-    fields = (
-        "input",
-        "measurement_start_time",
-        "probe_asn",
-        "probe_cc",
-        "report_id",
-        "test_name",
-        "test_start_time",
+    # TODO: use assertions for unknown bugs
+    rid = msm.get("report_id", "")
+    url = "https://explorer.ooni.org/measurement/{}".format(rid) if rid else "no rid"
```

Comment:
This is a good idea.

You probably also want to check for `input` and if so put it in the URL too.

Reply:
The idea here is to slowly build up a corpus of known bugs that we detect and ignore, and log out new ones.